src/main_thread/init/media_source_content_initializer.ts

```diff
@@ -708,6 +712,34 @@ export default class MediaSourceContentInitializer extends ContentInitializer {
             ? !playbackObserver.getIsPaused()
             : autoPlay;
           onReloadOrder({ position, autoPlay: autoplay });
+        };
+
+        switch (freezeResolution.type) {
+          case "reload": {
+            log.info("Init: Planning reload due to freeze");
+            triggerReload();
+            break;
+          }
+          case "flush": {
+            log.info("Init: Flushing buffer due to freeze");
+            const currentTime = observation.position.isAwaitingFuturePosition()
+              ? observation.position.getWanted()
+              : playbackObserver.getCurrentTime();
+            const relativeResumingPosition = freezeResolution.value.relativeSeek;
+            const wantedSeekingTime = currentTime + relativeResumingPosition;
+            playbackObserver.setCurrentTime(wantedSeekingTime);
+            break;
+          }
+          case "avoid-representations": {
+            const contents = freezeResolution.value;
+            if (this._initSettings.enableRepresentationAvoidance) {
+              manifest.addRepresentationsToAvoid(contents);
+            }
+            triggerReload();
```

Comment:
It may be possible but as we're in "weird bugs from devices" territory, I didn't want to try that uncertain flushing solution for now. Maybe in the future we may improve that freeze resolution if it also proves to work.

Reply:
Ok! make sense